.github/workflows/tag.yml

```diff
@@ -0,0 +1,93 @@
+name: Tag
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "The Infinispan branch to checkout when cutting the release."
+        required: true
+        default: "main"
+      version:
+        description: "Release version"
+        required: true
+      nextVersion:
+        description: "Next version"
+        required: true
+      dryRun:
+        description: "Dry run (doesn't push to Central)"
+        type: boolean
+        default: false
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: '23'
+          distribution: 'temurin'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+          gpg-private-key: ${{ secrets.INFINISPAN_MAVEN_GPG_ARMORED }}
+          gpg-passphrase: MAVEN_GPG_PASSPHRASE
+
+      - name: Checkout Source
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branch }}
+          ssh-key: ${{ secrets.INFINISPAN_RELEASE_PRIVATE_KEY }}
```

Comment:
why is the authentication needed?

Reply:
We need to commit, tag and push

Reply:
The GitHub token should have write permissions... right?
See
https://github.com/keycloak-rel/keycloak-rel/blob/62f7b2505a77abef1e9b36a130bf9b9084078d55/.github/workflows/x-create-tags.yml#L32-L38

Reply:
I copied from the protostream action.

Reply:
I've removed the `ssh-key` reference

Reply:
only works if the token has write permission set: https://github.com/ad-m/github-push-action?tab=readme-ov-file#requirements-and-prerequisites

Reply:
Then we're good to go